culinary-canvas/capstone-project/frontend/my_capstone_project/src/UserRecommendation.jsx

```diff
@@ -0,0 +1,110 @@
+import React, {useState} from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAuth } from './RedirectToAuthentication';
+import LoadingSpinner from './LoadingSpinner';
+
+function UserRecommendation(){
+    const navigate = useNavigate();
+    const {user, login} = useAuth();
+    const [cookingLevel, setCookingLevel] = useState('');
+    const [dietaryPreferences, setDietaryPreferences] = useState([]);
+    const [favoriteFoods, setFavoriteFoods] = useState('');
+    const [loading, setLoading] = useState(false)
+    const backendAddress =import.meta.env.VITE_BACKEND_ADDRESS
+
+    const handleCheckboxChange = (event) => {
+        const {value, checked} = event.target;
+        setDietaryPreferences((prev) => checked ? [...prev, value] : prev.filter((pref) => pref !== value))
+    };
+
+    const handleSubmit = async(event) => {
+        event.preventDefault();
+        setLoading(true)
+        if(user){
+            try{
+                const response =  await fetch(`${backendAddress}/save-preferences`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type' : 'application/json'
+                    },
+                    body: JSON.stringify({
+                        userId: user.id,
+                        cookingLevel,
+                        dietaryPreferences,
+                        favoriteFoods: favoriteFoods.split(',').map(food => food.trim()).filter(food => food !== '')}),
```

Comment:
Hmm, I don't think this is a good user-experience. 
1. It's not obvious to the user that the list of their favorite foods needs to be comma-separated.
2. I wonder how it would handle bad inputs, for example if the user were to type `Fish,,,Steak`, what would happen?

Instead, I think you leverage what you've already built here:
<img width="129" alt="Screenshot 2024-07-10 at 9 33 33 PM" src="https://github.com/OluwadaraO/Capstone-project-/assets/139017143/53f3c81e-b3fc-44db-b7d2-997d24c41475">

This approach, where textboxes are dynamically added, is best because it's less error-prone to string manipulation with commas etc.

Note: in both places, you should make sure that each value in those textboxes is not empty. That way the user can't click `Add favorite food` a hundred times with empty textboxes.
- Maybe also add a limit (e.g. 5 or 10 maximum favorite foods, again so the user can't overload the system).